src/main/java/com/salesforce/storm/spout/dynamic/config/ConfigDefinition.java

```diff
@@ -0,0 +1,126 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Defines a Configuration value.
+ * Highly inspired by Kafka's ConfigDef class.
+ */
+public class ConfigDefinition {
+    private final Map<String, ConfigKey> configKeys;
+
+    public ConfigDefinition() {
+        configKeys = new LinkedHashMap<>();
+    }
+
+    /**
+     * @return All of the keys used in the configuration.
+     */
+    public Set<String> getKeys() {
+        return Collections.unmodifiableSet(configKeys.keySet());
+    }
+
+    public ConfigDefinition define(ConfigKey key) {
+        if (configKeys.containsKey(key.name)) {
+            // TODO handle
+            //throw new ConfigException("Configuration " + key.name + " is defined twice.");
+        }
+        configKeys.put(key.name, key);
+        return this;
+    }
+
+    /**
+     * Utility to quickly define a new ConfigDef instance.
+     * @param name
+     * @param type
+     * @param defaultValue
+     * @param importance
+     * @param description
+     * @param category
+     * @return
+     */
+    public ConfigDefinition define(
+        String name,
+        Class type,
+        Object defaultValue,
+        Importance importance,
+        String category,
+        String description) {
+        return define(new ConfigKey(name, type, defaultValue, importance, description, category));
+    }
+
+    /**
+     * Get the configuration keys.
+     * @return a map containing all configuration keys
+     */
+    public Collection<ConfigKey> getConfigKeys() {
+        return Collections.unmodifiableCollection(configKeys.values());
+    }
+
+    public Object getDefaultValue(final String name) {
+        if (configKeys.containsKey(name)) {
+            return configKeys.get(name).defaultValue;
+        }
+        return null;
+    }
+
+    public boolean hasDefinedKey(final String name) {
+        return configKeys.containsKey(name);
+    }
+
+    /**
+     * The importance level for a configuration.
+     */
+    public enum Importance {
+        HIGH, MEDIUM, LOW
+    }
+
+    public static class ConfigKey {
+        public final String name;
+        public final Class type;
+        public final String description;
+        public final Object defaultValue;
+        public final Importance importance;
+        public final String category;
+
+        public ConfigKey(
+            final String name,
+            final Class type,
+            final Object defaultValue,
+            final Importance importance,
+            final String category,
+            final String description) {
+            this.name = name;
+            this.type = type;
+            this.defaultValue = defaultValue;
+            this.importance = importance;
+            this.description = description;
+            this.category = category;
+        }
+
+        public boolean hasDefault() {
+            return this.defaultValue != null;
+        }
+    }
+
+    public String toHtmlTable() {
```

Comment:
Just a stub for now, probably make this generate our readme, or we can move that to some other class.  Whatever works.